server/TracyEvent.hpp

```diff
@@ -530,36 +530,95 @@ struct CrashEvent
+        Win32_WrTerminated      = 22,
+        Win32_WrProcessInSwap   = 23,
+        Win32_WrCpuRateControl  = 24,
+        Win32_WrCalloutStack    = 25,
+        Win32_WrKernel          = 26,
+        Win32_WrResource        = 27,
+        Win32_WrPushLock        = 28,
+        Win32_WrMutex           = 29,
+        Win32_WrQuantumEnd      = 30,
+        Win32_WrDispatchInt     = 31,
+        Win32_WrPreempted       = 32,
+        Win32_WrYieldExecution  = 33,
+        Win32_WrFastMutex       = 34,
+        Win32_WrGuardedMutex    = 35,
+        Win32_WrRundown         = 36,
+        Win32_WrAlertByThreadId = 37,
+        Win32_WrDeferredPreempt = 38,
+        Win32_WrPhysicalFault   = 39,
+        Win32_WrIoRing          = 40,
+        Win32_WrMdlCache        = 41,
+        Win32_WrRcu             = 42,
+        Win32_MaximumWaitReason,
+    };
+
+    tracy_force_inline int64_t Start() const { return _start.Val(); }
+    tracy_force_inline void SetStart( int64_t start ) { assert( start < (int64_t)( 1ull << 47 ) ); _start.SetVal(start); }
+    tracy_force_inline int64_t End() const { return _end.Val(); }
+    tracy_force_inline void SetEnd( int64_t end ) { assert( end < (int64_t)( 1ull << 47 ) ); _end = end; }
+    tracy_force_inline bool IsEndValid() const { return _end.IsNonNegative(); }
+    tracy_force_inline uint8_t Cpu() const { return _cpu; }
+    tracy_force_inline void SetCpu( uint8_t cpu ) { _cpu = cpu; }
+    tracy_force_inline uint8_t WakeupCpu() const { return _wakeupcpu; }
+    tracy_force_inline void SetWakeupCpu( uint8_t wakeupcpu) { _wakeupcpu = wakeupcpu; }
+    tracy_force_inline CSReason Reason() const { return CSReason( _reason ); }
+    tracy_force_inline void SetReason( int8_t reason ) { _reason = reason; }
+    tracy_force_inline int8_t State() const { return _state; }
+    tracy_force_inline void SetState( int8_t state ) { _state = state; }
     tracy_force_inline int64_t WakeupVal() const { return _wakeup.Val(); }
     tracy_force_inline void SetWakeup( int64_t wakeup ) { assert( wakeup < (int64_t)( 1ull << 47 ) ); _wakeup.SetVal( wakeup ); }
     tracy_force_inline uint16_t Thread() const { return _thread; }
     tracy_force_inline void SetThread( uint16_t thread ) { _thread = thread; }
 
-    tracy_force_inline void SetStartCpu( int64_t start, uint8_t cpu ) { assert( start < (int64_t)( 1ull << 47 ) ); _start_cpu = ( uint64_t( start ) << 16 ) | cpu; }
-    tracy_force_inline void SetEndReasonState( int64_t end, int8_t reason, int8_t state ) { assert( end < (int64_t)( 1ull << 47 ) ); _end_reason_state = ( uint64_t( end ) << 16 ) | ( uint64_t( reason ) << 8 ) | uint8_t( state ); }
+    tracy_force_inline void SetStartCpu( int64_t start, uint8_t cpu ) { assert( start < (int64_t)( 1ull << 47 ) ); _start.SetVal( start ); _cpu = cpu; }
+    tracy_force_inline void SetEndReasonState( int64_t end, int8_t reason, int8_t state ) { assert( end < (int64_t)( 1ull << 47 ) ); _end = end; _reason = reason; _state = state; }
```

Comment:
These functions existed only to have the setting operation as a single store. With the way data is now split it makes no sense to keep these.